processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryEngine.java

```diff
@@ -337,7 +338,9 @@ public static CursorBuildSpec makeCursorBuildSpec(TimeseriesQuery query, @Nullab
                        .setAggregators(query.getAggregatorSpecs())
                        .setQueryContext(query.context())
                        .setPreferredOrdering(
-                           query.isDescending() ? Cursors.descendingTimeOrder() : Cursors.ascendingTimeOrder()
+                           query.getGranularity().equals(Granularities.ALL)
+                           ? ImmutableList.of()
```

Comment:
nit: i think we have been slowly migrating to using stuff like `List.of` instead of `ImmutableList` when possible